### ISSUE TITLE
weight parameters

### ISSUE THREAD

**Issue by igormis** [ThilinaRajapakse/simpletransformers#117]
I am doing Classification problem, and I have three labels, which are not balanced:
label 0 - 2632 samples
label 1 - 613 samples
and
label 2 - 426 samples,
I am passing 
`model = ClassificationModel('roberta', 'roberta-large', num_labels=3, args={'weight':[2632, 613, 426],'num_train_epochs': 10,'reprocess_input_data': True, 'overwrite_output_dir': True}) `
Is this OK, or I should use fraction?

**Comment by ThilinaRajapakse** [comment]
The safest way would be to use fractional weights. I'm not sure if it would mess things up if you don't, but you might see some terrifyingly large loss values.

See this [discussion](https://discuss.pytorch.org/t/passing-the-weights-to-crossentropyloss-correctly/14731/8) for more info. The weights are simply passed to the Pytorch loss function.

**Comment by igormis** [comment]
So this numbers would be ok:
- 0.71
- 0.17
- 0.12
?

**Comment by ThilinaRajapakse** [comment]
Should be, yeah.

**Comment by ThilinaRajapakse** [comment]
Or, keep the weight for `label 0` as 1, and build the weights for the others based on that.

**Comment by kinoute** [comment]
In my case I did:

```py
from sklearn.utils.class_weight import compute_class_weight
weights = compute_class_weight('balanced', np.unique(train_dataset.labels), train_dataset.labels)
weights
# array([1.02972652, 0.97194164])
```
Then

```py
from scipy.special import softmax
weights = softmax(weights).tolist()
weights
# [0.5144422009314233, 0.4855577990685766]
```

And I passed the weights variable to the ClassificationModel:

```py
model = ClassificationModel(train_dataset, weight=weights, args={...})
```

Not sure it's the right approach though.

**Comment by ThilinaRajapakse** [comment]
It's not wrong as far as I know, but the softmax isn't necessary. You can just use the weights from the first part.

```
from sklearn.utils.class_weight import compute_class_weight
weights = compute_class_weight('balanced', np.unique(train_dataset.labels), train_dataset.labels)
weights
# array([1.02972652, 0.97194164])
```

I think this is the best approach.
